lib/features/auth/cubit/auth_cubit.dart

```diff
@@ -180,11 +190,20 @@ class AuthCubit extends Cubit<AuthState> {
         amountLimit: country.isUS ? 4999 : 499,
       );
 
-      final unRegisteredUserExt = await _authRepositoy.registerUser(
+      var unRegisteredUserExt = await _authRepositoy.registerUser(
         tempUser: tempUser,
         isTempUser: true,
       );
 
+      final newNotificationId = await _updateNotificationId(
+        guid: unRegisteredUserExt.guid,
+        currentNotificationId: unRegisteredUserExt.notificationId,
```

Comment:
Maybe its better to fetch the notificationId from firebase instead of the old userext

Reply:
Did you check that method, what it does? I am using the old notificationId from the user to compare the one from the the firebase. If they are different it will be changed. Meaning that the user is either logging in with a different device or reinstalled the application

Reply:
Did you check the method as well? I am comparing the token from the userExtenstion and the one from firebase to see if he's logging in from a different device or reinstalled the app